operational_analysis/toolkits/pandas_plotting.py

```diff
@@ -710,3 +714,204 @@ def turbine_polar_contour(
         ax_carthesian.set_title("Turbine %s" % (tid))
 
     return ax_carthesian, ax_polar, artists, labels
+
+
+def luminance(rgb):
+
+    """Calculates the brightness of an rgb 255 color. See https://en.wikipedia.org/wiki/Relative_luminance
+
+    Args:
+        rgb(:obj:`tuple`): 255 (red, green, blue) tuple
+
+    Returns:
+        luminance(:obj:`scalar`): relative luminance
+
+    Example:
+
+        .. code-block:: python
+
+            >>> rgb = (255,127,0)
+            >>> luminance(rgb)
+            0.5687976470588235
+
+            >>> luminance((0,50,255))
+            0.21243529411764706
+
+    """
+
+    luminance = (0.2126 * rgb[0] + 0.7152 * rgb[1] + 0.0722 * rgb[2]) / 255
+
+    return luminance
+
+
+def color_to_rgb(color):
+
+    """Converts named colors, hex and normalised RGB to 255 RGB values
+
+    Args:
+        color(:obj:`color`): RGB, HEX or named color
+
+    Returns:
+        rgb(:obj:`tuple`): 255 RGB values
+
+    Example:
+
+        .. code-block:: python
+
+            >>> color_to_rgb("Red")
+            (255, 0, 0)
+
+            >>> color_to_rgb((1,1,0))
+            (255,255,0)
+
+            >>> color_to_rgb("#ff00ff")
+            (255,0,255)
+    """
+
+    if isinstance(color, tuple):
+        if max(color) > 1:
+            color = tuple([i / 255 for i in color])
+
+    rgb = matplotlib.colors.to_rgb(color)
+
+    rgb = tuple([int(i * 255) for i in rgb])
+
+    return rgb
+
+
+def plot_windfarm(
+    project,
+    tile_name="OpenMap",
+    plot_width=800,
+    plot_height=800,
+    marker_size=14,
+    kwargs_for_figure={},
+    kwargs_for_marker={},
+):
+
+    """Plot the windfarm spatially on a map using the Bokeh plotting libaray.
+
+    Args:
+        project(:obj:`plant object`): project to be plotted
+        tile_name(:obj:`str`): tile set to be used for the underlay, e.g. OpenMap, ESRI, OpenTopoMap
+        plot_width(:obj:`scalar`): width of plot
+        plot_height(:obj:`scalar`): height of plot
+        marker_size(:obj:`scalar`): size of markers
+        kwargs_for_figure(:obj:`dict`): additional figure options for advanced users, see Bokeh docs
+        kwargs_for_marker(:obj:`dict`): additional marker options for advanced users, see Bokeh docs. We have some custom behavior around the "fill_color" attribute. If "fill_color" is not defined, OpenOA will use an internally defined color pallete. If "fill_color" is the name of a column in the asset table, OpenOA will use the value of that column as the marker color. Otherwise, "fill_color" is passed through to Bokeh.
```

Comment:
I added some extra documentation for the "fill_color"  parameter, for which some custom behavior is defined on lines 867-898. @charlie9578 , could you please verify my description is correct and update it as appropriate.

Reply:
Hi @jordanperr, thats the correct description. Thanks for cleaning up the code and checking things.